src/main/java/com/minecolonies/client/gui/WindowBuildTool.java

```diff
@@ -546,8 +547,37 @@ private void moveDownClicked(Button button)
     private void rotateRightClicked(Button button)
     {
         rotation = (rotation + ROTATE_RIGHT) % POSSIBLE_ROTATIONS;
-        RotationHelper.rotate(Settings.instance.getSchematicWorld(), EnumFacing.UP, true);
-        updatePosition();
+        updateRotation(rotation);
+    }
+
+    /**
+     * Updates the rotation of the structure depending on the input.
+     * @param rotation the rotation to be set.
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)